### ISSUE TITLE
Performance degradation on each training epoch with Training UI feature used

### ISSUE THREAD

**Issue by akrauchanka** [deeplearning4j/deeplearning4j#2663]
I've setup spark job to training MLP network with Remote UI according to https://deeplearning4j.org/visualization#sparkui manual with 10 training epochs. During the process of training I was monitoring log and noticed, that subsequent epoch take twice more time than previous in average. Server to visualize training statistics was deployed to different ec2 instance, dl4j libs version used is 0.7.0.  Could you look into this issue, please?

Thanks!

**Comment by AlexDBlack** [comment]
Can you give us some more info so we can look into this and hopefully reproduce it more easily? Specifically, the following would be useful:
- Number and type of instances used (CPU, GPU, cores, memory etc); configuration (executors, executor memory)
- How you configured the UI (mainly the collection frequency - defaults to 1)
- Approximate network size (number/type of layers, and approx. total parameter count)

**Comment by AKrauchanka** [comment]
Hi Alex,

I've used amazon EMR with with GPU instances g2.2xlarge (1 NVIDIA GRID GPU with 1,536 CUDA cores and 4 GB of video memory):
- m3.xlarge master and 11 g2.2xlarge core instances (1 for spark driver, so there were 10 executors); driver memory - 4G, driver memory overhead - 6G, drive cores - 8; driver extra options -Dorg.bytedeco.javacpp.maxbytes=5368709120; executor memory - 4G, executor memory overhead - 6G, executor cores - 4; executor extra options -Dorg.bytedeco.javacpp.maxbytes=5368709120; spark.locality.wait = 0; spark default level of parallelism - 40;

- UI configured accordingly to https://deeplearning4j.org/visualization#sparkui manual with all the default params, UI server is deployed to different ec2 instance;

- there are three layers in netrwork: 
1. DenseLayer, nIn - 11, nOut - 1000;
2. DenseLayer, nIn - 1000, nOut - 100;
3. OutputLayer, nIN - 100 nOut - 1, activation - sigmoind, loss function - XENT;
with total parameters - 112201;

- minibatch size - 2000, averaging frequency - 10;

There is input with 200 000 000 samples, each sample is 11 float values and 1 float values for label.

If statistics collection is enabled split processing time increase from 2-3 sec to 14-16 sec at first and then during the first epoch rise up to 16 min.

Let me know if more info is needed. 
Thanks in advance!

**Comment by AKrauchanka** [comment]
Closed by mistake.

**Comment by AlexDBlack** [comment]
Thanks, that should be enough info. I'll see if I can reproduce it with that.

**Comment by raver119** [comment]
This particular issue was resolved long ago.
